pilot/pkg/model/xds_cache.go

```diff
@@ -164,6 +173,26 @@ func newLru(evictCallback simplelru.EvictCallback[string, cacheValue]) simplelru
 	return l
 }
 
+func (l *lruCache) Run(stop <-chan struct{}) {
+	go func() {
+		ticker := time.Tick(5 * time.Second)
+		for {
+			select {
+			case <-ticker:
+				l.mu.Lock()
+				for _, keyConfigs := range l.evictQueue {
+					l.clearIndexes(keyConfigs.key, keyConfigs.dependentConfigs)
+				}
+				l.evictQueue = l.evictQueue[:0] // this is to reuse the underlying memory
```

Comment:
minor concern: if we always reuse we may grow this unbounded if we have a big spike? Not sure how much of an issue that is. Is there some way to shrink the cap to some large max (10k or something)?

Reply:
There is a three indexed slice which can be use to reslice


https://stackoverflow.com/questions/27938177/golang-slice-slicing-a-slice-with-sliceabc